src/components/events.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StaticQuery, graphql } from 'gatsby';
+import { contentExists } from '../utils/ug-utils.js';
+import moment from 'moment';
+import '../styles/events.css';
+
+function findMatches(fromDrupal, fromWP) {
+    for (let i=0; i < fromDrupal.length; i++) {
+        for (let j=0; j < fromWP.length; j++) {
+            return fromDrupal[i].name === fromWP[j].name;
+        }
+    }
+}
+
+const generateEvents = (data, eventData) => {
+    let drupalCategories = eventData.relationships.field_event_category;
+    let matchAllCategories = eventData.field_match_categories;
+    let title = eventData.field_title;
+    let allEvents = data.allWpEvent.edges;
+    let shownEvents = [];
+    
+    if (matchAllCategories) {
+        for (let i=0; i < allEvents.length; i++) {        
+            if (JSON.stringify(drupalCategories) === JSON.stringify(allEvents[i].node.eventsCategories.nodes)) {
+                shownEvents.push(allEvents[i]);
+            }        
+        } 
+    } else {
+        for (let i=0; i < allEvents.length; i++) {
+            if (findMatches(drupalCategories, allEvents[i].node.eventsCategories.nodes)) {
+                shownEvents.push(allEvents[i]);
+            }
+        } 
+    }
+    
+    return (contentExists(shownEvents) ?
+    <React.Fragment key={eventData.drupal_id}>
+        <h2 className="mb-5">{contentExists(title) ? title : "Upcoming Events"}</h2>
+        <div className="event-list row gx-3 gy-5 justify-content-md-center mb-5">

```

Comment:
Can we make this a list instead of divs?

Reply:
When I initially tried that, the default UoG styles messed it up. However, now that I'm using some custom styles anyway, I might be able to override them.

Reply:
Done via CSS `!important` overrides